app/controllers/sufia/uploads_controller.rb

```diff
@@ -73,6 +73,11 @@ def asset_attributes
         params.fetch(:generic_work, nil) || params.fetch(:batch_upload_item, nil)
       end
 
+      def duplicate_generic_work(fileset)
```

Comment:
I'm having trouble with one of the sufia uploads controller tests for it, working on that now. The rest of the changes, though, much better, thanks.